great_expectations/rule_based_profiler/helpers/simple_semantic_type_filter.py

```diff
@@ -98,20 +98,33 @@ def _build_table_column_name_to_inferred_semantic_domain_type_map(
         self,
         batch_ids: List[str],
         validator: Validator,
-        column_names: List[str],
+        column_names: Optional[List[str]] = None,
     ) -> None:
         column_types_dict_list: List[Dict[str, Any]] = validator.get_metric(
             metric=MetricConfiguration(
                 metric_name="table.column_types",
                 metric_domain_kwargs={
-                    "batch_id": batch_ids[-1],  # active_batch_id
+                    "batch_id": validator.active_batch_id,
                 },
                 metric_value_kwargs={
                     "include_nested": True,
                 },
             )
         )
 
+        if column_names is None:
+            column_names = validator.get_metric(
```

Comment:
Just wanting to check: are we actually testing this code path? so far I've only been able to reach it in `non-FDS` and we pass in `column_names` explicitly

Reply:
:rocket: